app/controllers/partners/providers_controller.rb

```diff
@@ -56,27 +56,27 @@ def account_params
   def user_params
     {
       signup_type: partner.signup_type,
-      password: params[:password].presence || SecureRandom.hex,
-      email: params[:email],
-      first_name: params[:first_name],
-      last_name: params[:last_name],
+      password: params.require(:password).presence || SecureRandom.hex,
```

Comment:
If the password is not present we will use `SecureRandom.hex`.

If you use `require` it will raise an error which destroys the logic here. 
You need to permit it and check it presence.